src/frequenz/sdk/actor/data_sourcing/data_sourcing.py

```diff
@@ -0,0 +1,79 @@
+"""The DataSourcing Actor.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+
+import logging
+from typing import Dict, Optional
+
+from frequenz.channels import Receiver
+
+from frequenz.sdk.actor import actor
+from frequenz.sdk.actor.channel_registry import ChannelRegistry
+from frequenz.sdk.microgrid import ComponentCategory, microgrid_api
+
+from .common import ComponentMetricRequest
+from .microgrid_source import MicrogridApiSource
+
+
+@actor
+class DataSourcingActor:
+    """An actor that provides data streams of metrics as time series."""
+
+    def __init__(
+        self,
+        request_receiver: Receiver[ComponentMetricRequest],
+        registry: ChannelRegistry,
+    ) -> None:
+        """Create a `DataSourcingActor` instance.
+
+        Args:
+            request_receiver: A channel receiver to accept metric requests from.
+            registry: A channel registry.  To be replaced by a singleton
+                instance.
+        """
+        self._request_receiver = request_receiver
+        self._microgrid_api_source = MicrogridApiSource(registry)
+
+        self._comp_categories: Dict[int, ComponentCategory] = {}
+
+    async def get_component_category(self, comp_id: int) -> Optional[ComponentCategory]:
+        """Get the component category of the given component.
+
+        Args:
+            comp_id: Id of the requested component.
+
+        Returns:
+            The category of the given component, if it is a valid component, or
+            None otherwise.
+        """
+        if comp_id in self._comp_categories:
+            return self._comp_categories[comp_id]
+
+        api = microgrid_api.get().microgrid_api_client
```

Comment:
Why is the category resolution done at this level if it uses the microgrid API client? Shouldn't it be done by the `MicrogridApiSource` so the actor itself have no direct dependency to the microgrid API?

When getting data from parquet files at some point we'll need to check the category in some historical description of the microgrid too, right?

Reply:
I think the decision was that the parquet files will on the edge will get migrated when the component IDs change, etc.  So I think the microgrid api will be the source of this information for historical data as well.

Reply:
But I guess you're right,  the microgrid api won't be available when running the actor on the cloud to just read historical data.  I will move it into `MicrogridApiSource` tomorrow.

Reply:
done